addons/mail/static/src/utils/common/format.js

```diff
@@ -181,7 +181,7 @@ function generateMentionsLinks(body, { partners = [], threads = [] }) {
         const placeholder = `#-mention-channel-${thread.id}`;
         const text = `#${escape(thread.displayName)}`;
         mentions.push({
-            class: "o_channel_redirect",
+            class: "btn btn-sm btn-link align-baseline fw-normal px-1 py-0 o_channel_redirect",
```

Comment:
It's not ok to put style classnames here, because this is in the actual content of message, and we won't be able to change for previously sent messages.